synapse/models/infotech.py

```diff
@@ -1076,6 +1095,113 @@ def getModelDefs(self):
                         'doc': 'The file representing the value of the registry key, if the value is binary data.',
                     }),
                 )),
+
+                # TODO: all of the `id:dev:repo` forms need to be tied to the TBD inet:service model
+                ('it:dev:repotype', {}, ()),
+                ('it:dev:repo', {}, (
+                    ('name', ('str', {'lower': True, 'strip': True}), {
+                        'doc': 'The name of the repository.',
+                    }),
+                    ('desc', ('str', {}), {
+                        'disp': {'hint': 'text'},
+                        'doc': 'A free-form description of the repository.',
+                    }),
+                    ('created', ('time', {}), {
+                        'doc': 'When the repository was created.',
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'A URL where the repository is hosted.',
+                    }),
+                    ('upstream', ('it:dev:repo', {}), {
+                        'doc': 'The repository that changes are copied to or from.',
+                    }),
+                    ('type', ('it:dev:repotype', {}), {
+                        'doc': 'The type of the version control system used.',
+                        'ex': 'svn'
+                    }),
+                    ('submodules', ('array', {'type': 'it:dev:repo:commit'}), {
+                        'doc': "An array of other repos that this repo has as submodules, pinned at specific commits.",
+                    }),
+                )),
+
+                ('it:dev:repo:commit', {}, (
+                    ('repo', ('str', {}), {
+                        'doc': 'The repository the commit lives in.',
+                    }),
+                    ('parents', ('array', {'type': 'it:dev:repo:commit'}), {
+                        'doc': 'The commit or commits this commit is immediately based on.',
+                    }),
+                    ('branch', ('str', {}), {
```

Comment:
Is a branch a "thing" of it's own?

Reply:
I initially opted for no, since it made sense to me that a branch only becomes a real boy when a commit is applied to it and doesn't hold much weight of its own. But I'm rethinking that because of the value we could get of recording the branching pattern, some time props, and a few other things. 

I'll throw a new form up for an `it:dev:repo:branch` form. It feels like enough of a "thing" of its own.